src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyWithoutShouldStartLine.java

```diff
@@ -0,0 +1,63 @@
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyWithoutShouldStartLine {
+
+    private int a;
+    private static int b;
+    {  a = 2; }  //violation if option=ALONE
+    static { b = 3; }  //violation if option=ALONE
+
+    void method1() {
+        Thread t = new Thread() {@Override public void run() {
+            int a; int b;} //violation
+        };
+    }
+
+    void method2(java.util.HashSet<String> set) {
+        java.util.Map<String, String> map1 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+            put("alpha", "betical");}  //violation
+        };
+
+        java.util.Map<String, String> map2 = new java.util.LinkedHashMap<String, String>() {{
+            put("Hello", "World");
+            put("first", "second");
+            put("polygene", "lubricants");
+            put("alpha", "betical");
+        }}; //violation
+    }
+
+    void method3() {
+        method2(new java.util.HashSet<String>() {{
+            add("XZ13s");
+            add("AB21/X");
+            add("YYLEX");
+            add("AR5E");
+        }});  //violation
+    }
+
+    int method4(int a) {
+        if (a>2) a*=10; return ++a; }   //violation
+
+    void method5(int a) {
+        while (a > 5) { a--; }  //violation if option=ALONE
+
+        if (a > 4) { a++; }  //violation if option=ALONE
+
+        do {a--;} while (a > 3);  //violation
```

Comment:
This (_L#49:17_) violation happens for the option `alone_or_singleline`. This is similar to an existing violation at [InputRightCurlyAloneOrSingleline.java#L69:29](https://github.com/checkstyle/checkstyle/blob/master/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyAloneOrSingleline.java#L69) and also to [InputRightCurlyWithoutShouldStartLine.java#L53:27](https://github.com/ps-sp/checkstyle/blob/82161c80c0f9bcaffd08127f91ba7a401ea3d72f/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyWithoutShouldStartLine.java#L53).

Please confirm if such violations are acceptable.

Reply:
as for me, there should be no violation for `alone_or_singleline` as it is single block on line.

no violations at InputRightCurlyAloneOrSingleline.java#L69:29 and also to InputRightCurlyWithoutShouldStartLine.java#L53:27 
Whole idea of this mode to allow single line stuff, nothing is mentioned about multi blocks on the same line. If user is ok to keep some block single-lined , I do believe he is ok with multiple blocks too.

@rnveach , please confirm.

Reply:
Whole line is singleline, and option is set to alone or singleline. I don't expect a violation.

Reply:
Done.